src/main/java/io/github/majusko/pulsar/reactor/PulsarFluxConsumer.java

```diff
@@ -0,0 +1,235 @@
+package io.github.majusko.pulsar.reactor;
+
+import com.google.common.base.Strings;
+import io.github.majusko.pulsar.constant.Serialization;
+import io.github.majusko.pulsar.error.exception.ClientInitException;
+import org.apache.pulsar.client.api.SubscriptionType;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.util.concurrent.Queues;
+
+public class PulsarFluxConsumer<T> implements FluxConsumer<T> {
+    private final Sinks.Many<T> simpleSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+    private final Sinks.Many<FluxConsumerHolder> robustSink = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+    private final String topic;
+
+    private final Class<?> clazz;
+
+    private final Serialization serialization;
+
+    private final SubscriptionType subscriptionType;
+
+    private final String consumerName;
+
+    private final String subscriptionName;
+
+    private final int maxRedeliverCount;
+
+    private final String deadLetterTopic;
+
+    private final boolean simple;
+
+    // TODO add buffer size configuration
```

Comment:
It'll be better to just open an issue about it if you prefer doing that later

Reply:
Thank you. Forgot to search for TODOs 🤦‍♂️ before PR.

Reply:
https://github.com/majusko/pulsar-java-spring-boot-starter/issues/195